core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -86,20 +320,347 @@ public TableMetadata getTableMetadata(String namespace, String table) throws Exe
+                getGlobalIndexResourceID(namespace, table, secondaryIndex), scalingType));
+      }
+    }
+
+    // request
+    for (DeleteScalingPolicyRequest deleteScalingPolicyRequest : deleteScalingPolicyRequestList) {
+      try {
+        applicationAutoScalingClient.deleteScalingPolicy(deleteScalingPolicyRequest);
+      } catch (ApplicationAutoScalingException e) {
+        if (!(e instanceof ObjectNotFoundException)) {
+          LOGGER.warn(
+              "Delete scaling policy " + deleteScalingPolicyRequest.policyName() + " failed. " + e);
+        }
+      }
+    }
+
+    for (DeregisterScalableTargetRequest deregisterScalableTargetRequest :
+        deregisterScalableTargetRequestList) {
+      try {
+        applicationAutoScalingClient.deregisterScalableTarget(deregisterScalableTargetRequest);
+      } catch (ApplicationAutoScalingException e) {
+        if (!(e instanceof ObjectNotFoundException)) {
+          LOGGER.warn(
+              "Deregister scalable target "
+                  + deregisterScalableTargetRequest.resourceId()
+                  + " failed. "
+                  + e);
+        }
+      }
+    }
+  }
+
+  private RegisterScalableTargetRequest buildRegisterScalableTargetRequest(
+      String resourceID, String type, int ruValue) {
+    return RegisterScalableTargetRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAP.get(type))
+        .minCapacity(ruValue > 10 ? ruValue / 10 : ruValue)
+        .maxCapacity(ruValue)
+        .build();
+  }
+
+  private DeregisterScalableTargetRequest buildDeregisterScalableTargetRequest(
+      String resourceID, String type) {
+    return DeregisterScalableTargetRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAP.get(type))
+        .build();
+  }
+
+  private PutScalingPolicyRequest buildPutScalingPolicyRequest(String resourceID, String type) {
+    return PutScalingPolicyRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAP.get(type))
+        .policyName(getPolicyName(resourceID, type))
+        .policyType(PolicyType.TARGET_TRACKING_SCALING)
+        .targetTrackingScalingPolicyConfiguration(getScalingPolicyConfiguration(type))
+        .build();
+  }
+
+  private DeleteScalingPolicyRequest buildDeleteScalingPolicyRequest(
+      String resourceID, String type) {
+    return DeleteScalingPolicyRequest.builder()
+        .serviceNamespace(ServiceNamespace.DYNAMODB)
+        .resourceId(resourceID)
+        .scalableDimension(SCALABLE_DIMENSION_MAP.get(type))
+        .policyName(getPolicyName(resourceID, type))
+        .build();
+  }
+
+  protected static void waitForTableCreation(DynamoDbClient client, String tableFullName) {
```

Comment:
Can it be package private instead of `protected`?
(Any reason for protected?)

Reply:
It was used in `DynamoTableMetadataManager` so I made it protected.

Reply:
So, it can be package-private, right? 
I've checked the code and `DynamoTableMetadataManager` n the same package is the only one that uses this method except for this Admin class, so please make it package-private.
Restricted scope is usually better.

Reply:
Thank you. I removed the `protected` identifier [90037fd]( https://github.com/scalar-labs/scalardb/pull/301/commits/90037fdc8400d01e7387d17c33f4069dc9c02c9a)